### PR TITLE
Fix dependencies

### PR DESCRIPTION
**Pull request by doncem**
Last commit actually suppresses what's unfixable. The rest is step-by-step dependency review/update. Note: PMD plugin update is most attention requiring change hence the biggest diff on it

### CODE REVIEW

File: java-logging-httpcomponents/src/main/java/uk/gov/hmcts/reform/logging/httpcomponents/OutboundRequestLoggingInterceptor.java [hunk 1]
```diff
@@ -26,7 +26,7 @@ public OutboundRequestLoggingInterceptor() {
         this(systemUTC());
     }
 
-    OutboundRequestLoggingInterceptor(Clock clock) {
+    /* default */ OutboundRequestLoggingInterceptor(Clock clock) {
```
Comment (timja) [inline java-logging-httpcomponents/src/main/java/uk/gov/hmcts/reform/logging/httpcomponents/OutboundRequestLoggingInterceptor.java]: what's this for?

File: java-logging-spring/build.gradle [hunk 2]
```diff
@@ -1,5 +1,5 @@
 plugins {
-  id 'io.spring.dependency-management' version '1.0.6.RELEASE'
+  id 'io.spring.dependency-management' version '1.0.7.RELEASE'
```
Comment (timja) [inline java-logging-spring/build.gradle]: feels wrong the duplication but, you didn't introduce it...

File: java-logging-spring/build.gradle [hunk 3]
```diff
@@ -33,7 +33,7 @@ bintray {
 }
 
 def versions = [
-  springBoot: '2.0.8.RELEASE'
+  springBoot: '2.1.3.RELEASE'
```
Comment (timja) [inline java-logging-spring/build.gradle]: and here

File: java-logging-httpcomponents/src/main/java/uk/gov/hmcts/reform/logging/httpcomponents/OutboundRequestLoggingInterceptor.java (hunk 1 above)
Comment (doncem) [reply java-logging-httpcomponents/src/main/java/uk/gov/hmcts/reform/logging/httpcomponents/OutboundRequestLoggingInterceptor.java]: new pmd checks. all can be reviewed in a single commit ;)

File: java-logging-spring/build.gradle (hunk 2 above)
Comment (doncem) [reply java-logging-spring/build.gradle]: i know. just left in current structure

**Pull request merged**